src/v/cloud_storage/prefetch_tracker.cc

```diff
@@ -0,0 +1,72 @@
+#include "cloud_storage/prefetch_tracker.h"
+
+#include "cloud_storage/logger.h"
+#include "config/configuration.h"
+
+namespace cloud_storage {
+
+prefetch_tracker::prefetch_tracker()
+  : _enabled(config::shard_local_cfg().cloud_storage_prefetch_enable.value())
+  , _prefetch_threshold(
+      config::shard_local_cfg().cloud_storage_prefetch_threshold.value())
+  , _prefetch_size(
+      config::shard_local_cfg().cloud_storage_prefetch_size_trigger.value()) {}
+
+prefetch_tracker::prefetch_tracker(
+  bool enabled, size_t threshold, size_t limit) noexcept
+  : _enabled(enabled)
+  , _prefetch_threshold(threshold)
+  , _prefetch_size(limit) {}
+
+void prefetch_tracker::on_new_segment() {
+    _segment_ready = false;
+    _segment_bytes = 0;
+    _segment_size = 0;
+}
+
+void prefetch_tracker::set_segment_size(size_t sz) { _segment_size = sz; }
+
+void prefetch_tracker::on_bytes_consumed(size_t sz) {
+    _total_bytes += sz;
+    _segment_bytes += sz;
+}
+
+bool prefetch_tracker::operator()() {
```

Comment:
nit: could we give this method a descriptive name like `should_prefetch` or something?  Makes it easier to e.g. search for places its used compared to invoking it as `()`